terraform/main.tf

```diff
@@ -0,0 +1,72 @@
+provider "aws" {
+  region  = var.aws_region
+}
+
+resource "aws_security_group" "linc-cv-sg" {
+  name = "linc-cv-sec-group-${terraform.workspace}"
+  description = "Allow HTTP and SSH traffic via Terraform"
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+    ingress {
+    from_port   = 5000
+    to_port     = 5000
+    protocol    = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_instance" "ec2_instance" {
+  ami           = var.ami_id
+  instance_type = var.instance_type
+  key_name      = var.key_name
+  vpc_security_group_ids = [aws_security_group.linc-cv-sg.id]
+  root_block_device {
+    volume_size           = "20"
+    volume_type           = "gp2"
+    delete_on_termination = true
+  }
+
+  tags = {
+    Name = "Linc-CV-${terraform.workspace}-${formatdate("YYYY-MM-DD-hh:mm", timestamp() )}"
+  }
+    user_data = terraform.workspace == "prod" ? file("startup_script.sh") : file("startup_script_staging.sh")
+    user_data_replace_on_change = true
+}
+
+resource "aws_eip_association" "eip_assoc" {
+  count = terraform.workspace == "prod" ? 1 : 0
+  instance_id   = aws_instance.ec2_instance.id
+  allocation_id = "eipalloc-05ddad883ecfba225"
```

Comment:
What is `eipalloc-05ddad883ecfba225` and why are we hardcoding it?

Reply:
This is an elastic ip address. The EIP was deployed manually through the aws console so that it isn't destroyed with the terraform destroy command. This means that the prod instance will always have the same ip address and won't have to be changed each time we deploy. However, since we are going to be doing a green/blue deployment, we will now need two of the EIP's